### PR TITLE
Tyr: Add cosmogony2mimir to tyr

### PR DESCRIPTION
**Pull request by antoine-de**
This adds the capability for tyr to import [cosmogony](https://github.com/osm-without-borders/cosmogony/) data into mimir.

please review this PR carefully as I don't trust my `tyr` understanding :roll_eyes:

### CODE REVIEW

File: source/tyr/tyr/binarisation.py [hunk 1]
```diff
@@ -735,3 +735,27 @@ def ntfs2mimir(self, instance_config, input, job_id=None, dataset_uid=None):
...
+    """ launch cosmogony2mimir """
+    autocomplete_instance = models.db.session.merge(autocomplete_instance)  # reatache the object
+    logger = get_autocomplete_instance_logger(autocomplete_instance, task_id=job_id)
+    logger.debug('running cosmogony2mimir for {}'.format(job_id))
+    job = models.Job.query.get(job_id)
+    cnx_string = current_app.config['MIMIR_URL']
+    cosmo_file = unzip_if_needed(filename)
+
+    params = ['-i', cosmo_file, '--connection-string', cnx_string]
+    params.extend(['--dataset', autocomplete_instance.name])
```
Comment (kinnou02) [inline source/tyr/tyr/binarisation.py]: why not inline everything?

**Review by kinnou02 — APPROVED** [review]

File: source/tyr/tyr/binarisation.py [hunk 2]
```diff
@@ -735,3 +735,26 @@ def ntfs2mimir(self, instance_config, input, job_id=None, dataset_uid=None):
             models.db.session.commit()
 
         raise
+
+
+@celery.task(bind=True)
+def cosmogony2mimir(self, autocomplete_instance, filename, job_id, dataset_uid):
+    """ launch cosmogony2mimir """
+    autocomplete_instance = models.db.session.merge(autocomplete_instance)  # reatache the object
```
Comment (pbougue) [inline source/tyr/tyr/binarisation.py]: minor :gb: : "reat`t`ach" (no final `e` also)

File: source/tyr/migrations/versions/584590c53431_add_cosmogony.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    op.alter_column('autocomplete_parameter', 'street', new_column_name='_street')
+    op.alter_column('autocomplete_parameter', 'address', new_column_name='_address')
+
+    op.add_column('autocomplete_parameter', sa.Column('admin', source_admin))
+    op.add_column('autocomplete_parameter', sa.Column('poi', source_poi))
+    op.add_column('autocomplete_parameter', sa.Column('street', source_street))
+    op.add_column('autocomplete_parameter', sa.Column('address', source_address))
+
+    op.execute(
+        'update autocomplete_parameter set admin=_admin::source_admin, poi=_poi::source_poi, street=_street::source_street, address=_address::source_address;'
```
Comment (pbougue) [inline source/tyr/migrations/versions/584590c53431_add_cosmogony.py]: Ah, first time I see that cast :) (OK, noob in posgres)
Does it behave correctly if `_admin=COSMOGONY` ? We probably wanna secure those kinds of problems falling back to default value.

**Review by pbougue — COMMENTED** [review]
Did you test this locally? (Have you met our QA ? :smile: )
Maybe it's possible/relevant to add a test like this one https://github.com/CanalTP/navitia/blob/dev/source/tyr/tests/integration/dataset_test.py#L118

**Comment by antoine-de** [comment]
yes this has been manually checked, but not to the point that I'm confident with the code (I hate python for this... :roll_eyes: ).

I pushed on this PR the integration with cosmogony2cities + the handling of retrocompatible id in mimir, so I think the reviews needs to be updated.

**Review by kinnou02 — APPROVED** [review]
I will merge this once the release has been made

**Review by pbougue — APPROVED** [review]
OK, let's test it as soon as possible :)

**Pull request merged**